sdk/ai/ai-projects/review/ai-projects.api.md

```diff
@@ -0,0 +1,1988 @@
+    metadata: Record<string, string> | null;
+    model: string;
+    name: string | null;
+    object: "assistant";
+    response_format?: AgentsApiResponseFormatOptionOutput | null;
+    temperature: number | null;
+    tool_resources: ToolResourcesOutput | null;
+    tools: Array<ToolDefinitionOutput>;
+    top_p: number | null;
+}
+
+// @public
+export type AgentRunResponse = PromiseLike<ThreadRunOutput> & {
+    stream: () => Promise<AgentEventMessageStream>;
+};
+
+// @public
+export interface AgentsApiResponseFormat {
+    type?: ApiResponseFormat;
+}
+
+// @public
+export type AgentsApiResponseFormatMode = string;
+
+// @public
+export type AgentsApiResponseFormatModeOutput = string;
+
+// @public
+export type AgentsApiResponseFormatOption = string | AgentsApiResponseFormatMode | AgentsApiResponseFormat;
+
+// @public
+export type AgentsApiResponseFormatOptionOutput = string | AgentsApiResponseFormatModeOutput | AgentsApiResponseFormatOutput;
+
+// @public
+export interface AgentsApiResponseFormatOutput {
+    type?: ApiResponseFormatOutput;
+}
+
+// @public
+export type AgentsApiToolChoiceOption = string | AgentsApiToolChoiceOptionMode | AgentsNamedToolChoice;
+
+// @public
+export type AgentsApiToolChoiceOptionMode = string;
+
+// @public
+export type AgentsApiToolChoiceOptionModeOutput = string;
+
+// @public
+export type AgentsApiToolChoiceOptionOutput = string | AgentsApiToolChoiceOptionModeOutput | AgentsNamedToolChoiceOutput;
+
+// @public
+export interface AgentsNamedToolChoice {
+    function?: FunctionName;
+    type: AgentsNamedToolChoiceType;
+}
+
+// @public
+export interface AgentsNamedToolChoiceOutput {
+    function?: FunctionNameOutput;
+    type: AgentsNamedToolChoiceTypeOutput;
+}
+
+// @public
+export type AgentsNamedToolChoiceType = string;
+
+// @public
+export type AgentsNamedToolChoiceTypeOutput = string;
+
+// @public (undocumented)
+export interface AgentsOperations {
+    cancelRun: (threadId: string, runId: string, requestParams?: OptionalRequestParameters) => Promise<ThreadRunOutput>;
+    cancelVectorStoreFileBatch: (vectorStoreId: string, batchId: string, requestParams?: OptionalRequestParameters) => Promise<VectorStoreFileBatchOutput>;
+    createAgent: (model: string, options?: Omit<CreateAgentOptions, "model">, requestParams?: OptionalRequestParameters) => Promise<AgentOutput>;
+    createMessage: (threadId: string, options: ThreadMessageOptions, requestParams?: OptionalRequestParameters) => Promise<ThreadMessageOutput>;
+    createRun: (threadId: string, assistantId: string, options?: Omit<CreateRunOptions, "assistant_id">, requestParams?: OptionalRequestParameters) => AgentRunResponse;
+    createThread: (options?: AgentThreadCreationOptions, requestParams?: OptionalRequestParameters) => Promise<AgentThreadOutput>;
+    createThreadAndRun: (assistantId: string, options?: Omit<CreateAndRunThreadOptions, "assistant_id">, requestParams?: OptionalRequestParameters) => AgentRunResponse;
```

Comment:
`createThread` / `createThreadAndRun` - can this be combined into single methods and `run` being an argument / option in the `option` bag? I'd recommend reviewing the [guidelines](https://azure.github.io/azure-sdk/general_introduction.html) and the [typescript guidelines](https://azure.github.io/azure-sdk/typescript_introduction.html) for naming conventions

Reply:
These are following typespec defined https://github.com/azure/azure-rest-api-specs/tree/feature/azure-ai-projects/specification/ai/Azure.AI.Projects

Reply:
Has that TypeSpec been reviewed / merged yet? If not, when are y'all planning to get that merged?